content/actions/guides/building-and-testing-powershell.md

```diff
@@ -0,0 +1,228 @@
+
+To automate your testing with PowerShell and Pester, you can add a workflow that runs every time a change is pushed to your repository. In the following example, `Test-Path` is used to check that a file called `resultsfile.log` is present. 
+
+This example workflow file must be added to your repository's `.github/workflows/` directory:
+
+{% raw %}
+```yaml
+name: Test PowerShell on Ubuntu
+on: push
+
+jobs:
+  pester-test:
+    name: Pester test
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out repository code
+      uses: actions/checkout@v2
+    - name: Perform a Pester test from the command-line
+      shell: pwsh
+      run: Test-Path resultsfile.log | Should -Be $true
+    - name: Perform a Pester test from the Tests.ps1 file
+      shell: pwsh
+      run: |
+        Invoke-Pester Unit.Tests.ps1 -Passthru
+```
+{% endraw %}
+
+* `shell: pwsh` - Configures the job to use PowerShell when running the `run` commands.
+* `run: Test-Path resultsfile.log` - Check whether a file called `resultsfile.log` is present in the repository's root directory. 
+* `Should -Be $true` - Uses Pester to define an expected result. If the result is unexpected, then {% data variables.product.prodname_actions %} flags this as a failed test. For example:
+
+  ![Failed Pester test](/assets/images/help/repository/actions-failed-pester-test.png)
+
+* `Invoke-Pester Unit.Tests.ps1 -Passthru` - Uses Pester to execute tests defined in a file called `Unit.Tests.ps1`. For example, to perform the same test described above, the `Unit.Tests.ps1` will contain the following:
+  ```
+  Describe "Check results file is present" {
+      It "Check results file is present" {
+          Test-Path resultsfile.log | Should -Be $true
+      }
+  }
+  ```
+
+### PowerShell module locations
+
+The table below describes the locations for various PowerShell modules in each {% data variables.product.prodname_dotcom %}-hosted runner.
+
+|| Ubuntu | Mac | Windows |
+|------|-------|------|----------|
+|**PowerShell system modules** |`/opt/microsoft/powershell/7/Modules/*`|`/usr/local/microsoft/powershell/7/Modules/*`|`C:\program files\powershell\7\Modules\*`|
+|**PowerShell add-on modules**|`/usr/local/share/powershell/Modules/*`|`/usr/local/share/powershell/Modules/*`|`C:\Modules\*`|
+|**User-installed modules**|`/home/runner/.local/share/powershell/Modules/*`|`/Users/runner/.local/share/powershell/Modules/*`|`C:\Users\runneradmin\Documents\PowerShell\Modules\*`|
+
+### Installing dependencies
+
+{% data variables.product.prodname_dotcom %}-hosted runners have PowerShell 7 and Pester installed. You can use `Install-Module` to install additional dependencies from the PowerShell Gallery before building and testing your code. You can also cache dependencies to speed up your workflow. For more information, see "[Caching dependencies to speed up your workflow](/actions/automating-your-workflow-with-github-actions/caching-dependencies-to-speed-up-workflows)."
```

Comment:
I think we should mention somewhere that we shouldn't rely on pre-installed packages.
Yes, Pester is pre-installed on every image but on images, we always install the latest version and update image every week. It can bring breaking changes.

For example, some time ago Pester was updated from 4.x to 5.x and brought a lot of breaking changes. Some customers who relied on pre-installed version, were broken.

I think it it worth to invoke `Install-Module` command always. If module is already pre-installed, this command will do nothing. I would even suggest for customers to use `Install-Module` with `-MaximumVersion` argument.

Reply:
Thanks for the review @maxim-lobanov 🎉  I'll work on getting these updates added.